library/common/jni/jni_interface.cc

```diff
@@ -52,12 +75,15 @@ static void jvm_on_exit(void*) {
   jvm_detach_thread();
 }
 
-extern "C" JNIEXPORT jlong JNICALL
-Java_io_envoyproxy_envoymobile_engine_JniLibrary_initEngine(JNIEnv* env, jclass, jobject context) {
-  jobject retained_context = env->NewGlobalRef(context); // Required to keep context in memory
-  envoy_engine_callbacks native_callbacks = {jvm_on_engine_running, jvm_on_exit, retained_context};
-  // TODO(junr03): wire up once Android support lands.
-  envoy_logger logger = {nullptr, nullptr, nullptr};
+extern "C" JNIEXPORT jlong JNICALL Java_io_envoyproxy_envoymobile_engine_JniLibrary_initEngine(
+    JNIEnv* env, jclass, jobject on_start_context, jobject envoy_logger_context) {
+  jobject retained_on_start_context =
+      env->NewGlobalRef(on_start_context); // Required to keep context in memory
+  envoy_engine_callbacks native_callbacks = {jvm_on_engine_running, jvm_on_exit,
+                                             retained_on_start_context};
+
+  jobject retained_logger_context = env->NewGlobalRef(envoy_logger_context);
+  envoy_logger logger = {jvm_on_log, jvm_logger_release, retained_logger_context};
```

Comment:
We need to not set the logger if the platform didn't give us one. 

https://github.com/envoyproxy/envoy-mobile/pull/1405/files#diff-90bc04f1f5d0a531db783375205a81872a6bb7830871d24a01ea1678e6a8793bR47

Won't work, because the c++ code will still create a logger and push logs onto the callback. So if a user did not set a logger with setLogger, the end result right now is we're blackholing all the logs.

Reply:
I guess this is a miscomm from this comment: https://github.com/envoyproxy/envoy-mobile/pull/1405#discussion_r620640935

My understanding of your comment was that we wouldn't need to do this check because we will always get that context/lambda. Originally, I had this check but I can add it back and do another audit of how we pass down the lambda.